ui/lib/core/addon/components/secret-list-header.hbs

```diff
@@ -25,6 +25,80 @@
+            <LinkToExternal @route="secretsListRoot" class="tag is-borderless" data-test-new-pki-beta-button>
+              <Icon @name="arrow-left" />
+              Return to old PKI
+            </LinkToExternal>
+          {{else}}
+            {{#if this.shouldHidePkiBetaModal}}
+              <LinkTo @route="vault.cluster.secrets.backend.pki.overview" class="tag is-success is-borderless">
+                <Icon @name="key" />
+                New PKI UI available
+              </LinkTo>
+            {{else}}
+              <button
+                type="button"
+                class="tag is-success is-v-centered pki-beta-button"
+                {{on "click" (action (mut this.modalOpen) true)}}
+                data-test-old-pki-beta-button
+              >
+                <Icon @name="key" />
+                New PKI UI available
+              </button>
+              <Modal
+                @title="New PKI Beta"
+                @onClose={{action (mut this.modalOpen) false}}
+                @isActive={{this.modalOpen}}
+                @showCloseButton={{true}}
+              >
+                <section class="modal-card-body">
+                  The new PKI beta includes, among other things:
+                  <ul class="pki-beta-modal-list">
+                    <li>Multiple issuers can now be generated and managed in the UI.</li>
+                    <li>Non-disruptive root rotation: quickly rotate a root within the UI without disrupting issuance.</li>
+                    <li>UI upgrades: smoother configuration, an overview page, and more certificate information.</li>
+                  </ul>
+                  You will always be able to return to this version.
+                  <br />
+                  <br />
+                  You can also copy the following URL and bookmark it to always go to the beta version:
+                  <br />
+                  <code class="has-text-danger has-background-white-bis">
+                    http://localhost:8200/ui/vault/secrets/{{@model.id}}/pki/overview
+                  </code>
+                  <br />
+                  <br />
+                  <input
+                    type="checkbox"
+                    name="hide beta modal"
+                    checked={{this.hidePkiBetaModal}}
+                    onchange={{this.toggleHidePkiBetaModal}}
+                  />
+                  <label class="has-text-weight-semibold" for="hide-beta-modal">
+                    Don't show me this again
+                  </label>
+                </section>
+                <footer class="modal-card-foot modal-card-foot-outlined">
+                  <button type="button" class="button is-primary" {{on "click" this.transitionToNewPki}}>
+                    Go to beta
+                  </button>
+                  <button
+                    type="button"
+                    class="button is-secondary"
+                    {{on "click" (action (mut this.modalOpen) false)}}
```

Comment:
To separate concerns (since `action` handled _[a lot](https://guides.emberjs.com/release/upgrading/current-edition/action-on-and-fn/#toc_benefits-of-action-on-and-fn)_ in older versions of Ember), Octane no longer uses the helper in the template, now it's just used a decorator in the js file. 😄 
 
 Since an argument is being passed here, we'd want to use `fn`. [Ember ref](https://guides.emberjs.com/release/upgrading/current-edition/action-on-and-fn/#toc_getting-used-to-action-on-and-fn)
```suggestion
                    {{on "click" (fn (mut this.modalOpen) false)}}
```